includes/tracks/events/class-wc-settings-tracking.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * WooCommerce Settings Tracking
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds actions to track usage of WooCommerce Settings.
+ */
+class WC_Settings_Tracking {
+	/**
+	 * Singleton instance.
+	 *
+	 * @var WC_Settings_Tracking
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Whitelisted WooCommerce settings to potentially track updates for.
+	 *
+	 * @var array
+	 */
+	protected $whitelist = array();
+
+	/**
+	 * WooCommerce settings that have been updated (and will be tracked).
+	 *
+	 * @var array
+	 */
+	protected $updated_options = array();
+
+	/**
+	 * Instantiate the singleton.
+	 *
+	 * @return WC_Settings_Tracking Singleton instance.
+	 */
+	public static function instance() {
```

Comment:
This class implements a singleton instance but all the methods are called statically, including the action hooks. 

These classes should stick to the same pattern; either have instances and hook in using `$this`, or leave things static and dump the instance.

Reply:
Ok I see the `whitelist` variable now. I still think one pattern should be selected - the whitelist variable could be static if leaving everything else static.

cc @claudiosanches ?

Reply:
I agree, we should define one pattern here, maybe those classes even could have some interface to help keep the pattern.

Reply:
We introduced another singleton Tracks class here: https://github.com/woocommerce/woocommerce/pull/22879

Since an instance is required in a few cases (this one and the PR above), we should standardize on that pattern. Can we introduce that refactor in a new PR, or would you like to see it as part of this one?